### PR TITLE
Support timestamp in queryTable in python connector

### PR DESCRIPTION
**Pull request by linzhou-db**
- Support timestamp in load_as_pandas
- Support timestamp in load_as_spark

### CODE REVIEW

File: python/delta_sharing/delta_sharing.py [hunk 1]
```diff
@@ -73,17 +74,24 @@ def load_as_pandas(
...
+) -> "PySparkDataFrame":  # noqa: F821
     """
     Load the shared table using the given url as a Spark DataFrame. `PySpark` must be installed,
     and the application must be a PySpark application with the Apache Spark Connector for Delta
     Sharing installed.
 
     :param url: a url under the format "<profile>#<share>.<schema>.<table>".
     :param version: an optional non-negative int. Load the snapshot of table at version.
+    :param timestamp: an optional string. Load the snapshot of table at version corresponding
+      to the timestamp.
```
Comment (chakankardb) [inline python/delta_sharing/delta_sharing.py:94]: Should we add comment that at most one of version/timestamp should be specified?

File: python/delta_sharing/delta_sharing.py [hunk 2]
```diff
@@ -99,6 +107,8 @@ def load_as_spark(url: str, version: Optional[int] = None) -> "PySparkDataFrame"
     df = spark.read.format("deltaSharing")
     if version is not None:
         df.option("versionAsOf", version)
+    if timestamp is not None:
+        df.option("timestampAsOf", timestamp)
```
Comment (chakankardb) [inline python/delta_sharing/delta_sharing.py:111]: Question: If the caller specifies both, are we relying on the server to return error?

**Review by chakankardb — APPROVED** [review]

File: python/delta_sharing/delta_sharing.py (hunk 1 above)
Comment (linzhou-db) [reply python/delta_sharing/delta_sharing.py:94]: added.

File: python/delta_sharing/delta_sharing.py (hunk 2 above)
Comment (linzhou-db) [reply python/delta_sharing/delta_sharing.py:111]: there's a test case in test_delta_sharing.py L623, it's checked in the deltaSharingDataSource, still at the client side.

**Pull request merged**